lib/queue.rb

```diff
@@ -1,31 +1,68 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @size = 10
+    @store = Array.new(@size)
+    @front = 0
+    @back = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    # Check for full
+   if @front == (@back + 1) % @size
+      self.expand
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @size
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    if self.empty?
+      return nil
+    end
+
+    value = @store[@front]
+    @front = (@front + 1) % @size
+    return value
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return @store[@front]
   end
 
   def size
```

Comment:
Size should be the number of elements in the queue not the size of the underlying buffer.